examples/java-matter-controller/java/src/com/matter/controller/commands/common/MatterCommand.java

```diff
@@ -113,4 +118,40 @@ private void maybeSetUpStack() throws Exception {
   private void maybeTearDownStack() {
     // ToDo:We need to call DeviceController::Shutdown()
   }
+
+  public void setTestResult(String result) {
+    mTestResult = Optional.of(result);
+  }
+
+  public void expectSuccess(long timeout) {
+    expectResult("Success", timeout);
+  }
+
+  private void expectResult(String expectedResult, long timeout) {
+    long start = System.currentTimeMillis();
+    while (mTestResult.isPresent())
+      try {
+        if (System.currentTimeMillis() > (start + timeout)) {
+          throw new RuntimeException("timeout!");
+        }
+        Thread.sleep(100);
+      } catch (Exception ex) {
```

Comment:
please add a comment in the catch block on why this is ignored: what exception could happen here and why is it ok to silently ignore them?

Why not restrict this to `InterruptedException` for clarity?

Reply:
agree, it is better to use InterruptedException, just update, thank you!